Network/HTTP/Auth.hs

```diff
@@ -36,6 +36,11 @@ import Data.Char
 import Data.Maybe
 import Data.Word ( Word8 )
 
+#include <ghcplatform.h>
+#ifdef HaLVM_TARGET_OS
+#define MIN_VERSION_network(a,b,c) 1
```

Comment:
This feels messy, I think I'd prefer you put the conditions inline where it's needed.